### ISSUE TITLE
Implementation of Barlow Twins

### ISSUE THREAD

**Issue by Sherrylone** [vturrisi/solo-learn#109]
Hello, thanks for your great work!
I have some questions about Barlow Twins. I used the official code on Cifar-10, and  modified the conv1=nn.Conv2d(3, 64, kernel_size=3, stride=1, padding=2, bias=False), maxpool = nn.Identity(). Then, I got the  accuracy @top1: 89.57. Other parameters are followed by https://github.com/facebookresearch/barlowtwins.
Could you give some suggestion? many thanks!

**Comment by vturrisi** [comment]
Hey, thanks for the comment!

We also do the same modification here:
https://github.com/vturrisi/solo-learn/blob/096a2f1b804d9985b4d436ad965c3758fb350b81/solo/methods/base.py#L142-L144

For how many epochs did you train? We got those numbers training for 1000 epochs.
About parameters, you can check the parameters in `bash_files/pretrain/cifar/barlow.sh`. One key difference between our code and the original implementation is LARS. While they implement it, ours is copied from Nvidia Apex. The main difference is that it supports gradient clipping and also comes with a different default `eta `value, which you can set via the following flags in our repo:
```
--grad_clip_lars \
--eta_lars 0.02 \
```

We saw that this helps with smaller datasets (Cifar10, Cifar100 and Imagenet100), but default values for LARS are better for Imagenet (we were able to reproduce similar numbers for BYOL with the LARS setting that I showed, but it performed around 1-2 accuracy points worse for 100 epochs.

If I can help in anything else, let me know :)
